src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -862,7 +862,7 @@ void QgsRelationReferenceWidget::filterChanged()
 
   QgsFeature f;
   QgsFeatureIds featureIds;
-  QString filterExpression = mFilterExpression;
+  QString filterExpression = mFilterExpression.isEmpty() ? mFilterExpression : QStringLiteral( " ( %1 ) " ).arg( mFilterExpression );
```

Comment:
```suggestion
  if( !mFilterExpression.isEmpty() )
    mFilterExpression = QStringLiteral( " ( %1 ) " ).arg( mFilterExpression );
```

Reply:
or `mFilterExpression.prepend( " (" ).append( ") " );`

Reply:
I guess you mean (note the missing `m` prefix):

```cpp
  QString filterExpression = mFilterExpression;
  if( !filterExpression.isEmpty() )
    filterExpression = QStringLiteral( " ( %1 ) " ).arg( filterExpression );
```

Reply:
In case you meant to change the member `mFilterExpression`, I would say it is better to keep it untouched and wrap the parentheses only in the places where they are needed.

I have applied the change as in my previous comment.